scripts/test.sh

```diff
@@ -81,26 +102,118 @@ start_packager() {
+      rm "$log_temp"
+    fi
+    xcrun xcodebuild -scheme "$1" -configuration "$CONFIGURATION" -sdk iphonesimulator -destination name="iPhone 5s" build test 2>&1 | tee "$log_temp" | "$XCPRETTY" -c --no-utf --report junit --output build/reports/junit.xml || {
 	    EXITCODE=$?
         printf "*** Xcode Failure (exit code $EXITCODE). The full xcode log follows: ***\n\n"
-        cat "$LOGTEMP"
+        cat "$log_temp"
         printf "\n\n*** End Xcode Failure ***\n"
         exit $EXITCODE
     }
+    rm "$log_temp"
   else
-    xcodebuild -scheme "$1" -configuration "$CONFIGURATION" -sdk iphonesimulator -destination id="$dest" build test || {
+    xcrun xcodebuild -scheme "$1" -configuration "$CONFIGURATION" -sdk iphonesimulator -destination id="$IOS_SIM_DEVICE" build test || {
 	    EXITCODE=$?
         echo "*** Failure (exit code $EXITCODE). ***"
         exit $EXITCODE
-	}
+	  }
   fi
 }
 
+setup_ios_simulator() {
+  # - Ensure one version of xcode is chosen by all tools
+  
+  if [[ -z "$DEVELOPER_DIR" ]]; then
+    export DEVELOPER_DIR="$(xcode-select -p)"
+  fi
+  
+  # - Kill the Simulator to ensure we are running the correct one
+  
+  echo "Resetting simulator using toolchain from: $DEVELOPER_DIR"
+  
+  # quit Simulator.app
+  local deadline=$((SECONDS+5))
+  while pgrep -qx Simulator && [ $SECONDS -lt $deadline ]; do
+    osascript -e 'tell app "Simulator" to quit without saving' || true
+    sleep 0.25 # the 
+  done
+  
+  # stop CoreSimulatorService
+  launchctl remove com.apple.CoreSimulator.CoreSimulatorService 2>/dev/null || true
+  sleep 0.25 # launchtl can take a small moment to kill services
+  
+  # kill them with fire
+  while pgrep -qx Simulator com.apple.CoreSimulator.CoreSimulatorService; do
+    pkill -9 -x Simulator com.apple.CoreSimulator.CoreSimulatorService || true
+    sleep 0.05
+  done
+  
+  # - Prod `simctl` a few times to make sure it is ready to go
+  local deadline=$((SECONDS+5))
+  while [ -z "`xcrun simctl list devices 2>/dev/null`" ] && [ $SECONDS -lt $deadline ]; do
+    : # nothing to see here
+  done
+  
+  # - Choose a device, if it has not already been chosen
+  local deadline=$((SECONDS+5))
+  while [ -z "$IOS_SIM_DEVICE" ] && [ $SECONDS -lt $deadline ]; do
+    IOS_SIM_DEVICE=`ruby -rjson -e "puts JSON.parse(%x{xcrun simctl list devices --json})['devices'].each{|os,group| group.each{|dev| dev['os'] = os}}.flat_map{|x| x[1]}.select{|x| x['availability'] == '(available)'}.each{|x| x['score'] = (x['name'] == '$ios_sim_default_device_type' ? 1 : 0) + (x['os'] == '$ios_sim_default_ios_version' ? 1 : 0)}.sort_by!{|x| [x['score'], x['name']]}.reverse![0]['udid']"`
+    export IOS_SIM_DEVICE
+  done
+  if [ -z "$IOS_SIM_DEVICE" ]; then
+    echo "*** Failed to determine the iOS Simulator device to use ***"
+    exit 1
+  fi
+  
+  # - Reset the device we will be using
+  
+  xcrun simctl shutdown "$IOS_SIM_DEVICE" 1>/dev/null 2>/dev/null || true # sometimes simctl gets confused
```

Comment:
Why? This is just more explicit about what is happening. I am not aware of any performance differences, are you?